.github/workflows/rust.yml

```diff
@@ -180,3 +180,24 @@ jobs:
       - name: Deploy develop # unused
         if: github.ref == 'refs/heads/develop'
         run: bash scripts/deploy_docker.sh develop
+
+  autodeploy:
+    runs-on: ubuntu-latest
+    needs: [rustfmt, tests, deploy]
+    if: github.ref == 'refs/heads/main' || github.ref == 'refs/heads/develop' || startsWith(github.ref, 'refs/tags/')
+    steps:
+    - uses: actions/checkout@v2
+    - name: Deploy Mainnet
```

Comment:
I think that this name is not very descriptive. As merges into the `main` branch represent deployments into `staging`. The CGW is chain agnostic too, only the legacy was chain specific, but then again even back then, merges to `main` represented `staging` deployments.

Reply:
Your comments are welcome, indeed these could make the pipeline more understandable